server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java

```diff
@@ -263,15 +266,21 @@ private MTDocument getTranslationFromMT(MTDocument request,
     }
 
     private @Nullable String addMachineTranslationsToDoc(HDocument doc,
-            HLocale targetLocale, String projectSlug,
-            String versionSlug, ContentState saveState, boolean overwriteFuzzy,
-            String backendId) {
+                                                         HLocale targetLocale,
+                                                         String projectSlug,
+                                                         String versionSlug,
+                                                         ContentState saveState,
+                                                         boolean overwriteFuzzy,
+                                                         String backendId,
+                                                         AsyncTaskHandle taskHandle) {
         DocumentId documentId = new DocumentId(doc.getId(),
                 doc.getDocId());
         entityManager.clear();
         List<HTextFlow> textFlowsToTranslate =
                 getTextFlowsByDocumentIdWithConstraints(targetLocale,
                         documentId, overwriteFuzzy);
+        // Increase progress for non-translated items
+        taskHandle.increaseProgress(doc.getTextFlows().size() - textFlowsToTranslate.size());
```

Comment:
So this, um, is the number of text flows we skip because they don't need to be translated? Maybe we should introduce a local variable `textFlowsToSkip` to make it clearer.